File: web/src/types/stockFundamental.ts
/** Valuation & market data for an A-share individual stock. */
export interface StockFundamental {
  stock_code: string;
  trade_date: string;
  /** PE (TTM) */
  pe_ttm?: number | null;
  /** Price to Book */
  pb?: number | null;
  /** Total market cap (万元 CNY) */
  total_mv?: number | null;
  /** Circulating market cap (万元 CNY) */
  circ_mv?: number | null;
  /** Free float turnover rate (%) */
  turnover_rate_f?: number | null;
  /** Volume ratio (vs 5-day avg) */
  volume_ratio?: number | null;
  /** Total shares (万股) */
  total_share?: number | null;
  /** Float shares (万股) */
  float_share?: number | null;
  /** Basic EPS (元) — from latest income statement */
  eps?: number | null;
  /** ROE (%) — from latest income statement */
  roe?: number | null;
  /** Revenue YoY growth (%) */
  revenue_yoy?: number | null;
  /** Gross profit margin (%) */
  grossprofit_margin?: number | null;
  /** Net profit margin (%) */
  netprofit_margin?: number | null;
}

File: app/schemas/stock_fundamental.py
from datetime import date

from pydantic import BaseModel, ConfigDict


class StockFundamentalResponse(BaseModel):
    """A-share stock valuation & market data (latest record)."""

    model_config = ConfigDict(from_attributes=True)

    stock_code: str
    trade_date: date
    pe_ttm: float | None = None
    pb: float | None = None
    total_mv: float | None = None  # 万元 CNY
    circ_mv: float | None = None   # 万元 CNY
    turnover_rate_f: float | None = None
    volume_ratio: float | None = None
    total_share: float | None = None  # 万股
    float_share: float | None = None  # 万股

    # Enriched from stock_income (latest quarter)
    eps: float | None = None
    roe: float | None = None
    revenue_yoy: float | None = None
    grossprofit_margin: float | None = None
    netprofit_margin: float | None = None
